app/code/community/Razorpay/Payments/Model/Paymentmethod.php

```diff
@@ -90,12 +90,15 @@ public function validateSignature($response)
         $order = Mage::getModel('sales/order');
         $order->loadByIncrementId($session->getLastRealOrderId());
 
+        $amount = $order->getBaseGrandTotal();
+
         if ($this->hash_equals($signature , $response['razorpay_signature']))
         {
             $success = true;
             $order->sendNewOrderEmail();
             $order->setState(Mage_Sales_Model_Order::STATE_PROCESSING, true);
             $order->addStatusHistoryComment('Payment Successful. Razorpay Payment Id:'.$paymentId);
+            $order->setTotalPaid($amount);
```

Comment:
@captn3m0 So, you think we should instantiate an `api` object here and fetch the order and check the amount paid here?